Remove kuksa.val and VSC references

erikbosch opened the pull request:
This PR intends to do these things

- Replace the term "kuksa.val" with "kuksa"
- Remove references to old kuksa.val repositories
- Remove references to COVESA VSC as that project is dormant. COVESA has an IFEX project (which has taken over the VSC syntax but not the catalog) but Velocitas does not really have any connection to that 
- Also doing some minor updates to workflows to make it more visible what they actually do.

erikbosch committed:
Remove kuksa.val and VSC references

erikbosch commented on content/en/docs/concepts/development_model/val/_index.md:
```diff
@@ -96,6 +96,317 @@ Source code and build instructions are available in the respective KUKSA reposit
-## Guidelines
 
-* Guidelines for best practices on how to specify a gRPC-based service interface and on how to implement a vehicle service can be found in the [kuksa.val.services repository](https://github.com/eclipse/kuksa.val.services/tree/main/docs).
+## GRPC Interface Style Guide
```
Copying the info to here rather than referencing the file in the obsolete repo. Better to maintain the info here, if we want to keep it.

BjoernAtBosch replied:
Hmh, maybe worth to keep it. but if so, we should do this in a separate page.

BjoernAtBosch commented on content/en/docs/concepts/development_model/_index.md:
```diff
@@ -44,10 +44,6 @@ Vehicle data is stored in the KUKSA Databroker conforming to an underlying [Sema
-
-As an alternative to VSS and VSC, vehicle data and services can be defined semantically in a general IoT modelling language like Digital Twin Definition Language ([DTDL](https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/dtdlv2.md)) or BAMM Aspect Meta Model [BAMM](https://github.com/OpenManufacturingPlatform/sds-bamm-aspect-meta-model) as well.
-
 The Velocitas SDK is using [VSS](https://covesa.github.io/vehicle_signal_specification/) as the semantic model for the Vehicle Model.
```
Should we mention that it is possible to integrate (RPC-based) services based on protobuf/gRPC?

BjoernAtBosch commented on content/en/docs/concepts/development_model/vehicle_app_sdk/_index.md:
```diff
@@ -37,7 +37,7 @@ The Vehicle Model Ontology consists of the following classes:
 ### Model
 
-A model contains services, data points and other models. It corresponds to branch entries in VSS, interfaces in DTDL or namespaces in VSC.
+A model contains services, data points and other models.
```
```suggestion
A model contains data points (leaves) and other models (branches).
```

BjoernAtBosch reviewed (commented):
First findings

erikbosch committed:
Review findings

erikbosch commented:
@BjoernAtBosch  @lukasmittag - I have addressed the comments but also detect a few oddities.

- We have a workflow called CI but it only runs on merge
- We publish to "official documentation" only when running manual
- I do not really understand why we publish to two branches

I did some minor improvements to build scripts, but it should not affect anything practical.

BjoernAtBosch reviewed (approved):
LGTM
